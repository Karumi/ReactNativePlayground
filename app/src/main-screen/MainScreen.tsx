import { Card, CardItem, Container, Text } from "native-base";
import * as React from "react";
import { Image } from "react-native";
import Toolbar from "../base-components/toolbar/Toolbar";
import translator from "../i18n";
import screens from "../screens";
import assets from "./images/assets";

export interface Props {
    navigation: any;
}

class MainScreen extends React.Component<Props> {
    public render() {
        const { navigate } = this.props.navigation;
        const arrowForward = <Image
            style={{ width: 15, height: 15, position: "absolute", right: 10 }}
            source={assets.arrowForward} />;
        return (
            <Container>
                <Toolbar />
                <Card>
                    <CardItem button onPress={() => navigate(screens.resources)}>
                        <Text>{translator().resourcesScreenTitle}</Text>
                        {arrowForward}
                    </CardItem>
                    <CardItem button onPress={() => navigate(screens.navigation)}>
                        <Text>{translator().navigationScreenTitle}</Text>
                        {arrowForward}
                    </CardItem>
                    <CardItem button onPress={() => navigate(screens.list)}>
                        <Text>{translator().listScreenTitle}</Text>
                        {arrowForward}
                    </CardItem>
                    <CardItem button onPress={() => navigate(screens.scroll)}>
                        <Text>{translator().scrollScreenTitle}</Text>
                        {arrowForward}
                    </CardItem>
                    <CardItem button onPress={() => navigate(screens.grid)}>
                        <Text>{translator().gridScreenTitle}</Text>
                        {arrowForward}
                    </CardItem>
                    <CardItem button onPress={() => navigate(screens.camera)}>
                        <Text>{translator().cameraScreenTitle}</Text>
                        {arrowForward}
                    </CardItem>
                    <CardItem button onPress={() => navigate(screens.imagePicker)}>
                        <Text>{translator().imagePickerScreenTitle}</Text>
                        {arrowForward}
                    </CardItem>
                    <CardItem button onPress={() => navigate(screens.redux)}>
                        <Text>{translator().reduxScreenTitle}</Text>
                        {arrowForward}
                    </CardItem>
                    <CardItem button onPress={() => navigate(screens.layouts)}>
                        <Text>{translator().layoutsScreenTitle}</Text>
                        {arrowForward}
                    </CardItem>
                </Card>
            </Container>
        );
    }
}
export default MainScreen;
