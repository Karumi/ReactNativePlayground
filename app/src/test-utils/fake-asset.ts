jest.mock("../assets/asset", () => {
return {
        default: (uri) => uri,
    };
});
