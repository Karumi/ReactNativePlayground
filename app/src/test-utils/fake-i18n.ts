
import { I18nResolver } from "i18n-ts";
import en from "../translations/en";

const i18n = { default: en };

jest.mock("../i18n", () => {
return {
        default: () => new I18nResolver(i18n,  "en").translation,
    };
} );
