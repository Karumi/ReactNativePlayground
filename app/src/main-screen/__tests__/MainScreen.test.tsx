import * as React from "react";
import * as renderer from "react-test-renderer";
import "../../test-utils/fake-i18n";
import MainScreen from "../MainScreen";

jest.mock("../images/assets", () => {
    return {
        default: {
            arrowForward: "arrowForward",
        },
    };
});

jest.mock("../../base-components/toolbar/Toolbar", () => {
    return {
        default: "Toolbar",
    };
});

it("renders correctly with defaults", () => {
    const toolbar = renderer.create(<MainScreen navigation={jest.fn()}/>).toJSON();
    expect(toolbar).toMatchSnapshot();
});
