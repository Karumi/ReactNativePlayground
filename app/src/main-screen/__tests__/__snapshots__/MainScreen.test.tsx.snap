// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <Toolbar />
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        Resources
      </Text>
      <Image
        source="arrowForward"
        style={
          Object {
            "height": 15,
            "position": "absolute",
            "right": 10,
            "width": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        Navigation
      </Text>
      <Image
        source="arrowForward"
        style={
          Object {
            "height": 15,
            "position": "absolute",
            "right": 10,
            "width": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        List
      </Text>
      <Image
        source="arrowForward"
        style={
          Object {
            "height": 15,
            "position": "absolute",
            "right": 10,
            "width": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        Scroll
      </Text>
      <Image
        source="arrowForward"
        style={
          Object {
            "height": 15,
            "position": "absolute",
            "right": 10,
            "width": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        Grid
      </Text>
      <Image
        source="arrowForward"
        style={
          Object {
            "height": 15,
            "position": "absolute",
            "right": 10,
            "width": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        Camera
      </Text>
      <Image
        source="arrowForward"
        style={
          Object {
            "height": 15,
            "position": "absolute",
            "right": 10,
            "width": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        Image Picker
      </Text>
      <Image
        source="arrowForward"
        style={
          Object {
            "height": 15,
            "position": "absolute",
            "right": 10,
            "width": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        Redux
      </Text>
      <Image
        source="arrowForward"
        style={
          Object {
            "height": 15,
            "position": "absolute",
            "right": 10,
            "width": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        Layouts
      </Text>
      <Image
        source="arrowForward"
        style={
          Object {
            "height": 15,
            "position": "absolute",
            "right": 10,
            "width": 15,
          }
        }
      />
    </View>
  </View>
</View>
`;
